openapi-v1.yaml

```diff
@@ -6041,3 +6148,180 @@ paths:
           description: error response
           schema:
             $ref: "#/definitions/Error"
+
+  /groups/{namespace}:
+    parameters:
+      - name: namespace
+        type: string
+        in: path
+        required: true
+        description: The namespace to operate on
+    get:
+      description: Returns the attributes of one page of top level groups in namespace.
+      operationId: listTopLevelGroups
+      tags:
+        - groups
+      parameters:
+        - name: page
+          in: query
+          description: pagination offset
+          type: integer
+          required: false
+        - name: per_page
+          in: query
+          description: pagination limit
+          type: integer
+          required: false
+      responses:
+        200:
+          description: the group contents
+          schema:
+            $ref: '#/definitions/GroupListingTopLevel'
+        default:
+          description: error response
+          schema:
+            $ref: "#/definitions/Error"
+    post:
+      description: Creates a new, empty group in the namespace.
+      operationId: createGroup
+      tags:
+        - groups
+      parameters:
+        - in: body
+          name: body
+          schema:
+            $ref: '#/definitions/GroupCreate'
+      responses:
+        204:
+          description: group created successfully
+        default:
+          description: error response
+          schema:
+            $ref: "#/definitions/Error"
+
+  /groups/{namespace}/{name}:
+    parameters:
+      - name: namespace
+        type: string
+        in: path
+        required: true
+        description: The namespace of the group
+      - name: name
+        type: string
+        in: path
+        required: true
+        description: The name of the group
+    post:
```

Comment:
Maybe we can move this post under `/groups/{namespace}/{name}/{asset_namespace}/{asset_name}`?

Reply:
:+1:

Reply:
After the changes this comment sits at the wrong place. It suggested
`POST /groups/{namespace}/{name}/{asset_namespace}/{asset_name}` which is implemented. Resolving.